hamocc/mo_inventory_bgc.F90

```diff
@@ -0,0 +1,883 @@
+! terms of the GNU Lesser General Public License as published by the Free 
+! Software Foundation, either version 3 of the License, or (at your option) 
+! any later version. 
+!
+! BLOM is distributed in the hope that it will be useful, but WITHOUT ANY 
+! WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS 
+! FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for
+! more details. 
+!
+! You should have received a copy of the GNU Lesser General Public License 
+! along with BLOM. If not, see https://www.gnu.org/licenses/.
+
+
+module mo_inventory_bgc
+
+  use mo_carbch
+  use mo_sedmnt
+  use mo_biomod
+  use mo_control_bgc
+  use mo_bgcmean
+  use mo_param1_bgc
+  use mo_vgrid,       only: dp_min
+  use mod_xc,         only: ips, mnproc, nbdy, xcsum
+
+  implicit none
+  private
+
+  ! netCDF output file names
+  character(len=256), dimension(nbgcmax) :: fname_inv
+
+  ! ppm2con: atmospheric weight: ~10000kg/m^2, avrg. ~29 g/mol
+  ! --> 350 kmol/m^2 --> 1ppm ~ 0.35e-3 kmol/m^2
+  real, parameter :: ppm2con = 0.35e-3
+
+  real :: ztotvol
+  real :: ztotarea
+  ! aqueous sediment tracer
+  real :: zsedtotvol
+  real :: zpowtrato(npowtra)
+  ! non aqueous sediment tracer
+  real :: zsedhplto
+  real :: zsedlayto(nsedtra)
+  real :: zburial(nsedtra)
+  !  oceanic tracers
+  real :: zocetratot(nocetra)
+  real :: zocetratoc(nocetra)
+  ! additional ocean tracer
+  real :: zhito
+  real :: zco3to
+  ! alkalinity of the first layer
+  real :: zvoltop
+  real :: zalkali
+  ! atmosphere flux and atmospheric CO2
+  real :: zatmco2,zatmo2,zatmn2
+  real :: co2flux,so2flux,sn2flux,sn2oflux
+  real :: zprorca,zprcaca,zsilpro
+  real :: totalcarbon,totalphos,totalsil,totalnitr,totaloxy
+  ! sediment fluxes
+  real :: sum_zprorca
+  real :: sum_zprcaca
+  real :: sum_zsilpro
+  real :: sum_expoor
+  real :: sum_expoca
+  real :: sum_exposi
+  real :: sum_sedfluxo(npowtra)
+
+  public :: inventory_bgc
+
+contains
+
+SUBROUTINE INVENTORY_BGC(kpie,kpje,kpke,dlxp,dlyp,ddpo,WETO,iogrp)
+!*******************************************************************
+!
+!**** *INVENTORY_BGC* - calculate the BGC inventory.
+!
+!     P.Wetzel,              *MPI-Met, HH*    29.07.02
+!
+!     Modified
+!     --------
+!
+!     Purpose
+!     -------
+!     - calculate the BGC inventory.
+!
+!     Method
+!     -------
+!     -
+!
+!**   Interface.
+!     ----------
+!
+!     *CALL*       *INVENTORY_BGC*
+!
+!
+!**   Interface to ocean model (parameter list):
+!     -----------------------------------------
+!
+!
+!     Externals
+!     ---------
+!     none.
+!
+!**********************************************************************
+
+  implicit none
+
+  integer, intent(in) :: kpie,kpje,kpke
+  integer, intent(in) :: iogrp
+  real,    intent(in) :: dlxp(kpie,kpje)
+  real,    intent(in) :: dlyp(kpie,kpje)
+  real,    intent(in) :: ddpo(kpie,kpje,kpke)
+  real,    intent(in) :: WETO(kpie,kpje)
```

Comment:
we could rename WETO to omask for consistency with other routines

Reply:
Renamed WETO to omask.